xcm/xcm-builder/src/asset_conversion.rs

```diff
@@ -78,6 +78,32 @@ impl<
 		Ok((what, amount))
 	}
 }
+
+impl<
+		AssetId: Clone,
+		Balance: Clone,
+		ConvertAssetId: Convert<MultiLocation, AssetId>,
+		ConvertBalance: Convert<u128, Balance>,
+	> Convert<(AssetId, Balance), MultiAsset>
```

Comment:
@KiChjang 
it would be cool and useful to have backwards conversion  `(AssetId, Balance) -> MultiAsset` for `ConvertedConcreteId/MatchedConvertedConcreteId` automatically,

if I add the similar impl for `ConvertedConcreteId` for non-fungibles
```
impl<
		ClassId: Clone,
		InstanceId: Clone,
		ConvertClassId: Convert<MultiLocation, ClassId>,
		ConvertInstanceId: Convert<AssetInstance, InstanceId>,
	> Convert<(ClassId, InstanceId), MultiAsset>
 	for ConvertedConcreteId<ClassId, InstanceId, ConvertClassId, ConvertInstanceId>
```
it errors with:
```
error[E0119]: conflicting implementations of trait
```

so maybe, some new trait to extend traits `MatchesFungibles / MatchesNonFungibles` or some other solution?
or we dont want to support this here?

Reply:
Could you first give some context as to what problem this PR is trying to solve? It's hard to reason about your solutions here when I don't have much idea what's going on.

Reply:
ok, sure, no problem, I updated here desc: https://github.com/paritytech/polkadot/pull/6760#issue-1594771246

Reply:
Okay, but doesn't `reverse_ref` already does the thing that you expect it to?